### ISSUE TITLE
Page thumbnail is no longer loaded after restoring the tab using "Undo" button

### ISSUE THREAD

**Issue by softvision-oana-arbuzov** [mozilla-mobile/fenix#19308]
## Prerequisites
Multiple pages opened.

## Steps to reproduce
1. Open the browser.
2. Go to "Active tabs" list.
3. Remove one of the open tabs
4. Tap "Undo" button.
5. Observe the restored tab. 

### Expected behavior
Page thumbnails is loaded.

### Actual behavior
Page thumbnails is not loaded.

### Device information

* Device vendor / model and Android version: Firefox Nightly 210428 (🦎 90.0a1-20210426094327)
* Firefox for Android version: Google Pixel 5 (Android 11) - 1080 x 2340 pixels, 19.5:9 ratio (~432 ppi density)

### Notes:
1. Screenshot attached.
2. This was previously reported here: https://github.com/webcompat/web-bugs/issues/71580
![image](https://user-images.githubusercontent.com/12184325/116569446-10b90b00-a912-11eb-84c9-ad6d3b4ed3e1.png)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-10573)

**Comment by jonalmeida** [comment]
This was intentional behaviour that we implemented at the time, but I think we have the ability to fix this now within our `ThumbnailsMiddleware` by observing the `UndoAction`s instead.

**Comment by SoftVision-LorandJanos** [comment]
Hi @softvision-oana-arbuzov ! Thank you for your report. I managed to reproduce the issue described by you on the build you tested on (Nightly 210428), on the latest Nightly (210512), on the latest Beta (89.0.0 beta.6) and on the latest Release (88.1.3). I noticed that if the pages are not accessed and loaded all the way, the thumbnails are not loaded either, even if the page is accessed for the first time, and the same result is with reopening closed tabs, until they are accessed and completely loaded.

Devices used: Motorola Nexus 6 and HTC Nexus 9, both with Android 7.1.1.
